src/main/java/com/google/devtools/build/lib/exec/AbstractSpawnStrategy.java

```diff
@@ -105,15 +105,17 @@ public boolean canExec(Spawn spawn, ActionContext.ActionContextRegistry actionCo
     final Duration timeout = Spawns.getTimeout(spawn);
     SpawnExecutionContext context =
         new SpawnExecutionContextImpl(spawn, actionExecutionContext, stopConcurrentSpawns, timeout);
-    // TODO(ulfjack): Provide a way to disable the cache. We don't want the RemoteSpawnStrategy to
-    // check the cache twice. Right now that can't happen because this is hidden behind an
-    // experimental flag.
+
     SpawnCache cache = actionExecutionContext.getContext(SpawnCache.class);
     // In production, the getContext method guarantees that we never get null back. However, our
     // integration tests don't set it up correctly, so cache may be null in testing.
     if (cache == null) {
       cache = SpawnCache.NO_CACHE;
     }
+    // If we are using the RemoteSpawnRunner, it handles caching internally so avoid caching logic here.
+    if (spawnRunner.getName().equals("remote")) {
```

Comment:
Open to suggestions about this, feels a little hacky.

Reply:
Maybe add another method to the SpawnRunner interface?